libs/api-mocks/msw/handlers.ts

```diff
@@ -223,6 +223,53 @@ export const handlers = makeHandlers({
 
     return 204
   },
+  floatingIpCreate({ body, query }) {
+    const project = lookup.project(query)
+    errIfExists(db.floatingIps, { name: body.name })
+
+    const newFloatingIp: Json<Api.FloatingIp> = {
+      id: uuid(),
+      project_id: project.id,
+      ip: `${[...Array(4)].map(() => Math.floor(Math.random() * 256)).join('.')}`,
+      ...body,
+      ...getTimestamps(),
+    }
+    db.floatingIps.push(newFloatingIp)
+    return json(newFloatingIp, { status: 201 })
+  },
+  floatingIpList({ query }) {
+    const project = lookup.project(query)
+    const ips = db.floatingIps.filter((i) => i.project_id === project.id)
+    return paginated(query, ips)
+  },
+  floatingIpView({ query, path }) {
+    const project = lookup.project(query)
+    const ip = db.floatingIps.filter(
+      (i) => i.project_id === project.id && i.name === path.floatingIp
+    )[0]
+    return ip
+  },
+  floatingIpDelete({ path, query }) {
+    const floatingIp = lookup.floatingIp({ ...path, ...query })
+    db.floatingIps = db.floatingIps.filter((i) => i.id !== floatingIp.id)
+
+    return 204
+  },
+  floatingIpAttach({ path, query, body }) {
+    const floatingIp = lookup.floatingIp({ ...path, ...query })
+    const instance = lookup.instance({ ...path, ...query, instance: body.parent })
+    floatingIp.instance_id = instance.id
+
+    return floatingIp
+  },
```

Comment:
This should probably 400 if the IP already has an `instance_id`. It looks like that is the API behavior:

https://github.com/oxidecomputer/omicron/blob/1f48138840644e2573800b8f52dd00af35acc179/nexus/tests/integration_tests/external_ips.rs#L687-L704

I think the easy way to get that 400 is literally

```ts
if (floatingIp.instance_id) {
  throw "floating IP cannot be attached to one instance while still attached to another"
}
```

That will definitely return an error, and I'm like 95% sure it will automatically get a 400 status.